Index: minijail/common.mk
===================================================================
--- minijail.orig/common.mk
+++ minijail/common.mk
@@ -125,7 +125,7 @@ export SRC ?= $(CURDIR)
 # If BASE_VER is not set, read the libchrome revision number from
 # common-mk/BASE_VER file.
 ifeq ($(strip $(BASE_VER)),)
-BASE_VER := $(shell cat $(SRC)/../common-mk/BASE_VER)
+BASE_VER := "0.0.1"
 endif
 $(info Using BASE_VER=$(BASE_VER))
 
